HARK/core.py

```diff
@@ -327,6 +329,226 @@ def __str__(self):
     def describe(self):
         return self.__str__()
 
+    def del_param(self, param_name):
+        """
+        Deletes a parameter from this instance, removing it both from the object's
+        namespace (if it's there) and the parameters dictionary (likewise).
+
+        Parameters
+        ----------
+        param_name : str
+            A string naming a parameter or data to be deleted from this instance.
+            Removes information from self.parameters dictionary and own namespace.
+
+        Returns
+        -------
+        None.
+        """
+        if param_name in self.parameters:
+            del self.parameters[param_name]
+        if hasattr(self, param_name):
+            delattr(self, param_name)
```

Comment:
is there any way to avoid this redundant storage of the constructs?

Maybe constructs should not be stored in parameters, but in self.constructs (or something)

Reply:
That's for *deleting* things. Right now, I've written things to be intentionally redundant, because we're "between two worlds". The medium run goal is to remove the top level namespace at all, only putting them in `parameters`.

I don't want a separate `constructs` dictionary, because there's no meaningful distinction between `parameters` and `constructs`. It might be that `parameters` was the wrong name for that dictionary.